### ISSUE TITLE
if guard on `os.environ` doesn't "correct" type of `os.getenv`

### ISSUE THREAD

**Issue by ashb** [microsoft/pyright#1488]
**Describe the bug**

```python
if 'REPOS' in os.environ:
    REPO_CONFIGURATION = json.loads(os.getenv('REPOS'))
```

The `json.loads` line gives a warning about types:

```
Argument of type "str | None" cannot be assigned to parameter "s" of type "str | bytes" in function "loads"
  Type "str | None" cannot be assigned to type "str | bytes"
    Type "None" cannot be assigned to type "str | bytes"
      Type "None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "bytes"PyrightreportGeneralTypeIssues
```

**Expected behavior**

This is possibly an edge case, but since the `if X in os.environ` line, we know that os.getenv cannot return None in this case.

**VS Code extension or command-line**

VC Codium extension v 1.1.110

**Comment by erictraut** [comment]
A type checker cannot make assumptions like this. The conditional expression in your code is testing `os.environ`, but the body of the `if` statement is using a call expression `os.getenv()`.

The type-friendly way to write this code is:
```python
repos = os.getenv("REPOS")
if repos:
    REPO_CONFIGURATION = json.loads(repos)
```

**Comment by ashb** [comment]
A type checker _could_ in this specific case for `os.envrion`/`os.getenv`, but fair enough if you don't want to :)

**Comment by erictraut** [comment]
This would violate a key principle in pyright. The type checker bases its analysis on the type system and type information provided to it, not on specialized, hard-coded knowledge about the semantics of specific types and methods.
